chatGPT_API/Model/DomainModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  DomainModel.swift
+//  chatGPT_API
+//
+//  Created by Ai Hawok on 14/03/2023.
+//
+// API request is implemented here
+import Foundation
+
+protocol StoryManagerDelegate{
+    func didUpdateStory(story: String)
+}
+
+final class DomainModel{
+    var content = ""
+    var delegate : StoryManagerDelegate?
+    
+    private let apiKey = ConfigManager.shared.getAPIKey()
+    let url = "https://api.openai.com/v1/chat/completions"
+    
+    func fetch(with query: String){
```

Comment:
Another common way of communicating with your networking layer is through `completionHandlers`:
```swift
func fetch(with query: String, onCompletion: (Result<String, Error>) -> ()) {

}

fetch(with: query) { result in
    switch result {
    case .success(let story): // do something with story
    case .failure(let error): // do something with error
    }
}
```